### PR TITLE
PRO-28 Wallet app to support testnets

### PR DESCRIPTION
**Pull request by hemadri-tank-bacancy**
* Points
 1. Removed deprecated chain network
 2. Added new test chain networks (Görli, Sokol, Optimistic Görli etc...)
 3. Update harcoded token list for testnets
 4. Sending, Sending, NFTs

limitation: Show rates only for native tokens

https://user-images.githubusercontent.com/78720269/203771454-fadc638b-a5a1-48c1-9009-850891146773.mp4

### CODE REVIEW

File: src/configs/envConfig.js [hunk 1]
```diff
@@ -76,12 +76,12 @@ const envVars = {
...
+    TX_DETAILS_URL_BINANCE: 'https://testnet.bscscan.com/tx/',
     TX_DETAILS_URL_POLYGON: 'https://mumbai.polygonscan.com/tx/',
-    TX_DETAILS_URL_XDAI: 'https://blockscout.com/xdai/mainnet/tx/',
+    TX_DETAILS_URL_XDAI: 'https://blockscout.com/poa/sokol/tx/',
     TX_DETAILS_URL_AVALANCHE: 'https://testnet.snowtrace.io/tx/',
-    TX_DETAILS_URL_OPTIMISM: 'https://kovan-optimistic.etherscan.io/tx/',
+    TX_DETAILS_URL_OPTIMISM: 'https://blockscout.com/optimism/goerli/tx/',
     NETWORK_PROVIDER: 'kovan',
     COLLECTIBLES_NETWORK: 'rinkeby',
     OPEN_SEA_API: 'https://rinkeby-api.opensea.io/api/v1',
```
Comment (IAmKio) [inline src/configs/envConfig.js:87]: These three are deprecated also, just wondering where these are used?

```suggestion
    NETWORK_PROVIDER: 'kovan',
    COLLECTIBLES_NETWORK: 'rinkeby',
```

File: src/utils/tokens/binance-testnet-tokens.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[
+  {
+    "chain": "binance",
+    "address": "0x0000000000000000000000000000000000000000",
+    "name": "BNB",
+    "symbol": "BNB",
+    "decimals": 18,
+    "iconUrl": "https://images.prismic.io/pillar-app/d8e73973-82f1-4b20-a068-fe20e21816b0_0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c_1.png?auto=compress,format"
+  }
+]
```
Comment (IAmKio) [inline src/utils/tokens/binance-testnet-tokens.json:10]: Remember new line here

File: src/utils/tokens/ethereum-goerli-tokens.json [hunk 3]
```diff
@@ -9,154 +9,130 @@
...
-    "name": "Pillar",
-    "symbol": "PLR",
+    "address": "0x8e9192D6f9d903b1BEb3836F52a9f71E05846e42",
+    "name": "Sai Stablecoin v1.0",
+    "symbol": "SAI",
     "decimals": 18,
-    "iconUrl": "https://assets.coingecko.com/coins/images/809/thumb/PLR_Token.png?1595237935"
+    "iconUrl": "https://raw.githubusercontent.com/compound-finance/token-list/master/assets/asset_SAI.svg"
   }
 ]
```
Comment (IAmKio) [inline src/utils/tokens/ethereum-goerli-tokens.json:138]: New line for consistency

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just a few really minor points but other than that - it's looking good!

File: src/configs/envConfig.js (hunk 1 above)
Comment (hemadri-tank-bacancy) [reply src/configs/envConfig.js:87]: OPEN_SEA_API is used to get NFTs and NFTs transaction history.

File: src/utils/tokens/ethereum-goerli-tokens.json (hunk 3 above)
Comment (hemadri-tank-bacancy) [reply src/utils/tokens/ethereum-goerli-tokens.json:138]: Goerli Tokens fetched from Etherspot Playground. here need any changes?

File: src/utils/tokens/binance-testnet-tokens.json (hunk 2 above)
Comment (hemadri-tank-bacancy) [reply src/utils/tokens/binance-testnet-tokens.json:10]: means?

File: src/configs/envConfig.js (hunk 1 above)
Comment (IAmKio) [reply src/configs/envConfig.js:87]: So I noticed that Kovan and Rinkeby are still defined as values here, but these networks are deprecated. Do you know where we're using NETWORK_PROVIDER and COLLECTIBLES_NETWORK?

File: src/utils/tokens/binance-testnet-tokens.json (hunk 2 above)
Comment (IAmKio) [reply src/utils/tokens/binance-testnet-tokens.json:10]: Screenshot_20221207-075618

See screenshot

File: src/utils/tokens/ethereum-goerli-tokens.json (hunk 3 above)
Comment (IAmKio) [reply src/utils/tokens/ethereum-goerli-tokens.json:138]: Please see the above screenshot, a new line is needed at the end of the file

File: src/configs/envConfig.js (hunk 1 above)
Comment (hemadri-tank-bacancy) [reply src/configs/envConfig.js:87]: Usage for NETWORK_PROVIDER 
  1. check is production or staging status
  2. key wallet in find provider
  3. archanova in for environmentNetwork
  4. For network in user properties
 
 Usage for COLLECTIBLES_NETWORK
   1. In transferERC721 
   2. System information
   3. use in networkProvider for wallet instance connect
   
Should I need to update this?

File: src/utils/tokens/binance-testnet-tokens.json (hunk 2 above)
Comment (hemadri-tank-bacancy) [reply src/utils/tokens/binance-testnet-tokens.json]: Added a new line in last

File: src/utils/tokens/ethereum-goerli-tokens.json (hunk 3 above)
Comment (hemadri-tank-bacancy) [reply src/utils/tokens/ethereum-goerli-tokens.json]: https://github.com/pillarwallet/pillarwallet/pull/3174/commits/6f395eb16feb5a46f52e7f8a0be34fe304557c21

File: src/configs/envConfig.js (hunk 1 above)
Comment (IAmKio) [reply src/configs/envConfig.js:87]: I think let's leave this for later

**Review by IAmKio — APPROVED** [review]
LGTM

**Pull request merged**